src/main/java/org/openeo/spring/api/JobsApiController.java

```diff
@@ -379,78 +429,222 @@ public ResponseEntity<?> debugJob(
 			@Min(1) @Parameter(description = "This parameter enables pagination for the endpoint and specifies the maximum number of elements that arrays in the top-level object (e.g. jobs or log entries) are allowed to contain. The only exception is the `links` array, which MUST NOT be paginated as otherwise the pagination links may be missing ins responses. If the parameter is not provided or empty, all elements are returned.  Pagination is OPTIONAL and back-ends and clients may not support it. Therefore it MUST be implemented in a way that clients not supporting pagination get all resources regardless. Back-ends not supporting  pagination will return all resources.  If the response is paginated, the links array MUST be used to propagate the  links for pagination with pre-defined `rel` types. See the links array schema for supported `rel` types.  *Note:* Implementations can use all kind of pagination techniques, depending on what is supported best by their infrastructure. So it doesn't care whether it is page-based, offset-based or uses tokens for pagination. The clients will use whatever is specified in the links with the corresponding `rel` types.") @Valid @RequestParam(value = "limit", required = false) Integer limit) {
 		LogEntries logEntries = new LogEntries();
-		//TODO describe query
-		String elasticSearchQuery = "filebeat-7.13.3-2021.07.13-000001/_search";
+		
+		// Construcing ES query URL on selected index
+		String elasticSearchQuery = serviceIndexName + "/_search?size=10000"; //TODO (Lorenzo): sostituire
+		// il size con il metodo di paging indicato dalla documentazione (serve però anche, per facilitare 
+		// questo, riscrivere la query sotto direttamente tramite oggetto JSON anziché StringBuilder)
+				
 		try {
-			//TODO query elastic search endpoint here for all log information regarding a job queued for processing.
-			URL url = new URL(elasticSearchEndpoint + "/" + elasticSearchQuery);
-			HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-			conn.setRequestMethod("GET");
-			conn.setDoOutput(true);
+			
+	        HttpURLConnection conn = null;
+			URL url = new URL(elasticSearchEndpoint + elasticSearchQuery);
+			int check_code = HttpURLConnection.HTTP_OK;
+			
+			//check if elasticSearchEndpoint is an HTTP or HTTPS URL
+			if (elasticSearchEndpoint.startsWith("https://")) // break point here (10 aug 23)
+			{
+				HostnameVerifier customHostnameVerifier = new HostnameVerifier() {
+				    @Override
+				    public boolean verify(String hostname, SSLSession session) {
+				        // In questa implementazione, considereremo qualsiasi nome di host valido.
+				        // Se hai esigenze specifiche di verifica, puoi aggiungere la logica qui.
+				        return true;
+				    }
+				};
+
+				// Imposto come default questo verificatore personalizzato HTTPS temporaneo (TOFIX)
+				HttpsURLConnection.setDefaultHostnameVerifier(customHostnameVerifier);
```

Comment:
Do we need an always-true hostname verifier to allow the Elastic Search server, or doesn't the pre-existing default one work for us?

Reply:
postpone